### ISSUE TITLE
Nikto: occasionally failing scanner results in invalid json file

### ISSUE THREAD

**Issue by moxli** [secureCodeBox/secureCodeBox#2153]
### Describe the bug
I am using SecureCodeBox to scan a wide range of endpoints every week.
Each Saturday there are ~50 scans executed in a six hour period (distributed randomly).
Each week there are multiple (10-15) scans which fail. The pod actually finishes successfully, however the scanner in the nikto container actually runs into an error.
I was only able to catch the failing scans at a later point because the DefectDojo hook fails to import the results file of these scans.

I have looked at the container logs from the last 30 days and the error is **not** triggered for the same endpoints every time. Throughout the weeks scans for most endpoints have failed two or three times.

### Steps To Reproduce
1. Create a ScheduledScan utilizing Nikto
2. Sometimes / most of the time the scanner fails 
3. See the error in the nikto container log

### Expected behavior
I expect the Nikto scanner not to fail scanning the target and I expect the nikto container to quit with non-zero exit code in this case.
I also expected a check, if the provided json file is valid, in the parser and / or the persistence provider.

### System (please complete the following information):
 - secureCodeBox Version/Release: 4.3.0
 - Kubernetes Version: 

### Screenshots / Logs

Nikto Log
```
Defaulted container "nikto" out of: nikto, lurker
- Nikto v2.5.0
---------------------------------------------------------------------------
+ Target IP:          REDACTED
+ Target Hostname:    REDACTED
+ Target Port:        443
---------------------------------------------------------------------------
+ SSL Info:        Subject:  /CN=REDACTED
                   Altnames: REDACTED
                   Ciphers:  TLS_AES_256_GCM_SHA384
                   Issuer:   /C=US/O=Google Trust Services LLC/CN=GTS CA 1D4
+ Start Time:         2024-01-02 13:10:02 (GMT0)
---------------------------------------------------------------------------
+ Server: Google Frontend
+ /: Retrieved via header: 1.1 google.
Can't locate object method "new" via package "JSON" at /nikto/plugins/nikto_report_json.plugin line 102.
```

nikto-results.json (downloaded from S3 storage), notice how the file is invalid json
```
{"host":"REDACTED","ip":"REDACTED","port":"443","banner":"Google Frontend","vulnerabilities":[
```

findings.json (downloaded from S3 storage), empty brackets
```
[]
```

DefectDojo Persistence Log
```
2023-12-30 20:50:29 INFO  DefectDojoPersistenceProvider:25 - Starting DefectDojo persistence provider
2023-12-30 20:50:29 INFO  DefectDojoPersistenceProvider:36 - Downloading Scan Result
2023-12-30 20:50:30 INFO  DefectDojoPersistenceProvider:40 - Uploading Findings to DefectDojo at: https://REDACTED
2023-12-30 20:50:30 INFO  VersionedEngagementsStrategy:86 - Running with DefectDojo User Id: 4
2023-12-30 20:50:30 INFO  VersionedEngagementsStrategy:210 - Looking for ID of ProductType 'Customer Facing Infrastructure'
2023-12-30 20:50:30 INFO  VersionedEngagementsStrategy:217 - Using ProductType Id: 2
Exception in thread "main" io.securecodebox.persistence.defectdojo.exception.PersistenceException: Failed to attach findings to engagement.
        at io.securecodebox.persistence.defectdojo.service.DefaultImportScanService.createFindings(DefaultImportScanService.java:126)
        at io.securecodebox.persistence.defectdojo.service.DefaultImportScanService.reimportScan(DefaultImportScanService.java:78)
        at io.securecodebox.persistence.strategies.VersionedEngagementsStrategy.run(VersionedEngagementsStrategy.java:106)
        at io.securecodebox.persistence.DefectDojoPersistenceProvider.main(DefectDojoPersistenceProvider.java:43)
```

### Additional context
N/A

**Comment by moxli** [comment]
If we would be able to catch the error within the container, we might be able to workaround this issue for a while.
In this case the container could stop with a non-zero exit code and the scan would be restarted.

**Comment by J12934** [comment]
Hi @moxli 
i think this might be an issue in the version of nikto we are using.
I think this should be solved by switching to the acutal 2.5.0 release: https://github.com/sullo/nikto/pull/807

We are (if i remeber this right, feel free to correct me on this @Ilyesbdlala :D) using the nikto version from the [2.5.0 branch](https://github.com/sullo/nikto/tree/nikto-2.5.0).
Our (or at least my understanding) there was that nikto just considered everything on that branch to be version 2.5.0, like like everything on main was considered to be version 2.1.6.

We'll planning to switch to the proper 2.5.0 nikto version as soon as possible, but there are apparently changes to the json output format which might make this non trivial.

**Comment by moxli** [comment]
@J12934 Perfect thank you :)

**Comment by moxli** [comment]
@J12934 What do you think about verifying the json file before trying to import it? 
Or at least provide a better error message, because it took me a while to understand that the json file was the problem.

Also out of curiosity, do you recall what the reason for this change was?
https://github.com/secureCodeBox/secureCodeBox/commit/072bf3bb0c45006b92e11860deabed525ad6b1fc

Does nikto not exit with code 0 per default?

**Comment by moxli** [comment]
@J12934 There was another potentially breaking change to the json format, which is not a part of the 2.5.0 release:
https://github.com/sullo/nikto/pull/808

I asked the involved people if it breaks parsers again :) 

Also as far as I can tell DefectDojo does not support the new format yet, we might run into issues there.

**Comment by moxli** [comment]
I created an issue in the DefectDojo project to add support for the new json format:
https://github.com/DefectDojo/django-DefectDojo/issues/9274

**Comment by moxli** [comment]
@J12934 DefectDojo has added support for the new format here: https://github.com/DefectDojo/django-DefectDojo/issues/9274

Nikto 2.5.0 should not be used currently since only master produces valid json reports.
However the maintainer uses master as the latest working version.
https://github.com/sullo/nikto/pull/808#issuecomment-1887760408

Should this issue be used to add support for the new format?

**Comment by J12934** [comment]
Hi @moxli 
yeah we should switch back to the main / master branch of nikto.
Will need to slightly cahnge the parser, but should hopefully be not too bad. 
Opened up a bit more specifc issue #2220 to do the migration and prioritized it so that we should get a stable nikto version into the next release.

**Comment by moxli** [comment]
This has been fixed in #2220 :)
